code/modules/mob/living/basic/farm_animals/sheep.dm

```diff
@@ -0,0 +1,65 @@
+/mob/living/basic/sheep
+	name = "sheep"
+	desc = "Known for their soft wool and use in sacrifical rituals. Big fan of grass."
+	icon = 'icons/mob/sheep.dmi'
+	icon_state = "sheep"
+	icon_state = "sheep"
+	icon_dead = "sheep_dead"
+	gender = FEMALE
+	mob_biotypes = MOB_ORGANIC | MOB_BEAST
+	speak_emote = list("baas","bleats")
+	speed = 1.1
+	see_in_dark = 6
+	butcher_results = list(/obj/item/food/meat/slab = 3)
+	response_help_continuous = "pets"
+	response_help_simple = "pet"
+	response_disarm_continuous = "gently pushes aside"
+	response_disarm_simple = "gently push aside"
+	response_harm_continuous = "kicks"
+	response_harm_simple = "kick"
+	attack_verb_continuous = "kicks"
+	attack_verb_simple = "kick"
+	attack_sound = 'sound/weapons/punch1.ogg'
+	attack_vis_effect = ATTACK_EFFECT_KICK
+	health = 50
+	maxHealth = 50
+	gold_core_spawnable = FRIENDLY_SPAWN
+	blood_volume = BLOOD_VOLUME_NORMAL
+	ai_controller = /datum/ai_controller/basic_controller/sheep
+	var/mutable_appearance/cult_icon
+
+/mob/living/basic/sheep/Initialize(mapload)
+	AddComponent(/datum/component/mob_harvest, /obj/item/razor, /obj/item/food/grown/grass, /obj/item/stack/sheet/cotton/wool, "soft wool", 10, 3 MINUTES, 30 SECONDS, 5 SECONDS)
+	RegisterSignal(src, COMSIG_LIVING_HARVEST_UPDATE, .proc/wool_status)
+	. = ..()
+
+/mob/living/basic/sheep/proc/wool_status(atom/movable/source, amount_ready)
+	SIGNAL_HANDLER
+
+	if(amount_ready < 1)
+		icon_state = "sheep_harvested"
+		update_icon()
+	else
+		icon_state = "sheep"
+		update_icon()
+
+/mob/living/basic/sheep/update_overlays()
+	. = ..()
+	if(cult_icon)
+		. += cult_icon
+
+/mob/living/basic/sheep/proc/cult_time()
+	if(!cult_icon)
+		say("BAAAAAAAAH!")
+		cult_icon = image('icons/mob/sheep.dmi', "hat")
+		add_overlay(cult_icon)
+		update_overlays()
+
+
+/datum/ai_controller/basic_controller/sheep
```

Comment:
fixes: 
icon change signals not being sent
sacrificing the same lamb multiple times


```
	if(istype(sacrificial, /mob/living/basic/sheep))
		var/mob/living/basic/sheep/sacrificial_lamb = sacrificial
		if(sacrificial_lamb.cult_converted)
			for(var/cultists in invokers)
				to_chat(cultists, span_cultitalic("[sacrificial] has already been sacrificed!"))
				return FALSE
		for(var/cultists in invokers)
			to_chat(cultists, span_cultitalic("This feels a bit too cliché, don't you think?"))
		sacrificial_lamb.cult_time()
		return
```

```
	ai_controller = /datum/ai_controller/basic_controller/sheep
	var/cult_converted //were we sacrificed by cultists?
	var/ready_wool

/mob/living/basic/sheep/Initialize(mapload)
	AddComponent(/datum/component/mob_harvest, /obj/item/razor, /obj/item/food/grown/grass, /obj/item/stack/sheet/cotton/wool, "soft wool", 10, 3 MINUTES, 30 SECONDS, 5 SECONDS)
	RegisterSignal(src, COMSIG_LIVING_HARVEST_UPDATE, .proc/update_wool_icon)
	return ..()

/mob/living/basic/sheep/proc/update_wool_icon(atom/movable/source, amount_ready)
	SIGNAL_HANDLER
	ready_wool = amount_ready
	update_appearance(UPDATE_ICON)

/mob/living/basic/sheep/update_icon_state()
	. = ..()
	icon_state = "sheep[ready_wool < 1 ? "_harvested" : null]"

/mob/living/basic/sheep/update_overlays()
	. = ..()
	if(cult_converted)
		. += "hat"

/mob/living/basic/sheep/proc/cult_time()
	if(cult_converted)
		return
	cult_converted = TRUE
	say("BAAAAAAAAH!")
	update_appearance(UPDATE_ICON)

/datum/ai_controller/basic_controller/sheep
```